### PR TITLE
check status returned by ipatch_container_init_iter()

### LINKED ISSUE #48: Crash on cleaning up a DLS reader after trying to read a invalid file
Original report: https://github.com/LMMS/lmms/issues/5483
In the function `ipatch_dls_nullify_fixups`:
https://github.com/swami/libinstpatch/blob/a1af56bb064279f305aedf062597b700c2811f79/libinstpatch/IpatchDLSReader.c#L531-L534
`ipatch_container_init_iter` may fail and `inst_iter` will be invalid in that case. However, the function is using `inst_iter` without checking for errors.
In the reported case, FluidSynth tried to load a directory as a DLS file. However, it may happen for corrupted files as well.

### PR DESCRIPTION
**Pull request by jjceresa**
Fixes #48

### CODE REVIEW

File: libinstpatch/IpatchContainer.c [hunk 1]
```diff
@@ -347,7 +353,10 @@ ipatch_container_insert(IpatchContainer *container, IpatchItem *item, int pos)
     if(*child_types)	/* matching child type found? */
     {
         IPATCH_ITEM_WLOCK(container);
-        ipatch_container_init_iter(container, &iter, *child_types);
+        if(!ipatch_container_init_iter(container, &iter, *child_types))
+        {
+            return;
```
Comment (derselbst) [inline libinstpatch/IpatchContainer.c:358]: Missing `IPATCH_ITEM_WUNLOCK(container);`

File: libinstpatch/IpatchContainer.c [hunk 2]
```diff
@@ -438,7 +447,10 @@ ipatch_container_remove(IpatchContainer *container, IpatchItem *item)
         if(g_type_is_a(type, *child_types))   /* item type matches child type? */
         {
             IPATCH_ITEM_WLOCK(container);
-            ipatch_container_init_iter(container, &iter, *child_types);
+            if(!ipatch_container_init_iter(container, &iter, *child_types))
+            {
+                return;
```
Comment (derselbst) [inline libinstpatch/IpatchContainer.c:452]: IPATCH_ITEM_WUNLOCK(container);

File: libinstpatch/IpatchContainer.c [hunk 3]
```diff
@@ -528,7 +540,11 @@ ipatch_container_count(IpatchContainer *container, GType type)
         if(g_type_is_a(*child_types, type))   /* child type matches type? */
         {
             IPATCH_ITEM_RLOCK(container);
-            ipatch_container_init_iter(container, &iter, *child_types);
+            if(!ipatch_container_init_iter(container, &iter, *child_types))
+            {
+                return 0;
```
Comment (derselbst) [inline libinstpatch/IpatchContainer.c:545]: IPATCH_ITEM_RUNLOCK(container)

**Comment by PhysSong** [comment]
FYI, The reported crash is fixed in this PR.

**Comment by jjceresa** [comment]
Thanks for reviewing.

**Pull request merged**